chain/beacon/chain.go

```diff
@@ -41,12 +41,19 @@ type chainStore struct {
 func newChainStore(l log.Logger, cf *Config, cl net.ProtocolClient, c *cryptoStore, store chain.Store, t *ticker) *chainStore {
 	// we make sure the chain is increasing monotically
 	as := newAppendStore(store)
+
+	//
+	ss := newSchemeStore(as, cf.Group.Scheme)
```

Comment:
Actually I m wondering if we should not put it at a higher level like in ths [place](https://github.com/drand/drand/blob/ef80ae471878e58c5f9a30d9268ca7483d495886/core/drand_beacon.go#L258) in core/ where we create the bolt store: ths is _always_ needed unlike the other middlewares.
Or at the very least the only place we use a store outside of the beacon/ package is when we sync with a chain and we have a share yet, it's the "sync follow" CLI command [here](https://github.com/drand/drand/blob/master/core/drand_beacon_control.go#L1081) - we need to put that `SchemeStore` here as well - either directly to this line or where we create the boltstore first ([here](https://github.com/drand/drand/blob/ef80ae471878e58c5f9a30d9268ca7483d495886/core/drand_beacon.go#L258)) - not entirely sure what's best there yet...

Reply:
After reviewing it for a while, I am not sure we need to add the scheme store there. If you follow the code until followCmd func (the one that is called when you use follow chain cmd), you will see it doesn't know what the scheme is. Beacons received from other nodes will have prev signature on it or not depending on the scheme they use. The store will not run any checks over them (we are not using any wrapper on the store). Does this make sense?

Reply:
Ah I see - you're relying on the other node to give the right beacon. I would like to try to avoid that assumption on the other node - we dont know if he's honest or node. It's not a security issue but still having a consistent database format is the guarantee we should be able to provide, no matter what a node receives from the network.

Within the FollowChain command, the node ask the chain info of the remote note, so it seems relatively OK to add the restriction that the store follows the format according to the scheme, what do you think ?
```go
fstore, err := bp.createBoltStore(beaconID)
	if err != nil {
		bp.log.Errorw("", "beacon_id", beaconID, "start_follow_chain", "unable to create store", "err", err)
		return fmt.Errorf("unable to create store: %s", err)
	}
formatStore := chain.NewSchemeStore(fstore,info.Scheme)
```
I would have preferred to put it directly in `createBoltStore` but we don't necessarily know the scheme used at this point in this function.

Reply:
Ok, we can do that. I will add the extra code to do it then!